### PR TITLE
test: add property/fuzz tests for toon, secret redaction, CommonMark parsers

### LINKED ISSUE #399: [b2-mcp] Add property-based / fuzz tests for parsers (toon, secret redaction, CommonMark validator)
## Why
Example-based tests give high % coverage but miss adversarial edge cases. Three parsers are both security-relevant and edge-case-heavy, and one (the bespoke CommonMark validator) has been a recurring source of missed cases.

## Scope (use `fast-check`)
- **`src/utils/toon-encoder.ts`** — round-trip / structural properties across generated JSON inputs (nesting, arrays, special chars).
- **`src/utils/secret-sanitizer.ts`** — property: no injected secret/credential-shaped value survives redaction in any nested/array/header/error position; non-secret identifiers are preserved.
- **The privacy-page CommonMark validator** (`scripts/write-privacy-page.mjs`) — fuzz inputs to confirm it never emits unescaped/unsafe HTML and rejects unsupported constructs; ties into #384 (replace the bespoke parser).

## Acceptance
- [ ] fast-check property suites for the three parsers.
- [ ] At least one previously-uncaught edge case documented per parser (or a note that none was found).

Relates to #384 (durable CommonMark parser).

### PR DESCRIPTION
**Pull request by goanpeca**
## Summary

Adds `fast-check` property-based / fuzz test suites for the three security-relevant, edge-case-heavy parsers called out in #399, and fixes the CommonMark drift the fuzzer uncovered.

- **`src/utils/toon-encoder.ts`** — `tests/unit/toon-encoder.property.unit.test.ts`. Round-trips every generated JSON-compatible value through the canonical `@toon-format/toon` decoder (structural + scalar-type fidelity), and asserts lone surrogates always throw rather than emitting corrupt output.
- **`src/utils/secret-sanitizer.ts`** — `tests/unit/secret-sanitizer.property.unit.test.ts`. A canary secret injected in any nested / array / custom-header / error-message / error-field position never survives redaction (MCP + log modes), while non-secret identifiers (key IDs, bucket IDs) are preserved in MCP output.
- **Privacy-page CommonMark validator (`scripts/write-privacy-page.mjs`)** — `tests/unit/privacy-page.property.unit.test.ts`. Drives the existing differential CommonMark oracle: for any fuzzed input the validator must fail closed or produce HTML canonically equal to reference CommonMark (never `diverges`), and accepted output only ever contains the allow-listed tags.

All property suites use a checked-in, env-overridable seed (`tests/support/fast-check-seed.ts`) so each commit's pass/fail is deterministic and any failure is reproducible locally.

## Edge cases documented (acceptance criteria)

- **TOON encoder** — no divergence found across the committed 2000 runs (plus 5000+ ad-hoc runs during development) and every hand-picked corner (empty containers, whitespace-only strings, numeric/boolean-like strings, unsafe/empty keys, `-0`, `MAX_SAFE_INTEGER`, emoji graphemes, deep tabular/keyed shapes). Suite stands as a regression guard.
- **Secret sanitizer** — no secret escaped redaction in any generated position; suite is a regression guard.
- **CommonMark validator** — the fuzzer found two previously-uncaught families of "empty block" constructs the validator silently mis-rendered as paragraphs:
  1. bare list markers with no content (`+`, `*`, `1.`, `2)`) — empty list items in CommonMark; only `-` alone had been caught (by the thematic-break rule).
  2. bare ATX headings (`#`, `##`, `###`, `####`, `# `, `#\t`) — empty headings in CommonMark; the heading match required content, so hashes-only lines fell through to `<p>#</p>`.
  - Also caught: a top-level bullet whose content begins with another list marker (`- - x`, `- -`) is a nested list in CommonMark.
  - `scripts/write-privacy-page.mjs` was hardened to fail closed on all three, so it once again either rejects or matches reference CommonMark for every input. Ties into #384.

## Review fixes

- **Deterministic gate:** pinned a checked-in fast-check seed across all three suites and gave the two nested-list rejection branches distinct diagnostics (inline `- - x` vs. indented markers).
- **Finite doubles:** constrained the TOON `fc.double` generator to explicit finite bounds (`min`/`max` `Number.MAX_VALUE`, `noNaN`) so no `NaN`/`±Infinity` is generated and every value round-trips through JSON.
- **Seed robustness:** `FASTCHECK_SEED` now falls back to the default on an unset/empty value and throws on any non-integer, so a typo can never silently pass `0` or `NaN` to fast-check.
- Corrected the TOON suite comment so the stated input count matches the committed `numRuns`.

## Tests run

- `pnpm run typecheck` — clean
- `pnpm exec vitest --project=unit` — 1911 passed (incl. the three new suites)
- `pnpm exec vitest --project=contract` — passes after `pnpm run build` (doc-examples policy needs `dist/`); spelling policy clean
- `pnpm run build` — clean
- `biome check` on changed files — clean
- Privacy-page property suite run 5× (4000 runs each) with no divergence after the fix

## Notes / follow-ups

- `fast-check` added as a dev dependency; `deploy/customer-hosted/pnpm-lock.yaml` re-mirrored from the root lockfile to satisfy the package-surface policy.
- The CommonMark hardening relates to #384 (durable CommonMark parser) — these fixes reduce the drift surface until the bespoke parser is replaced.

Closes #399

### CODE REVIEW

File: tests/unit/toon-encoder.property.unit.test.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  fc.constantFrom("safe_key.1", "bad,key", " spaced ", "a{b}c", "true", "-", "", "value"),
+);
+
+const primitive = fc.oneof(
+  fc.constant(null),
+  fc.boolean(),
+  fc.integer(),
+  fc.double({ noNaN: true, noDefaultInfinity: true }),
+  leafString,
+);
```
[the remark this answers is not in the document]
Comment (goanpeca) [reply tests/unit/toon-encoder.property.unit.test.ts:72]: Fixed: constrained the double generator to explicit finite bounds (min/max Number.MAX_VALUE, noNaN) so no NaN or Infinity is generated and every value round-trips through JSON.

File: tests/support/fast-check-seed.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ * Deterministic default seed for merge-gating fast-check property suites.
+ *
+ * fast-check derives a fresh random seed per process by default, which makes a
+ * property suite a non-reproducible CI gate: a failure reports a seed buried in
+ * the log, and re-running picks a different seed so the failure may not recur.
+ * Pinning a checked-in default makes each commit's pass/fail deterministic and
+ * any failure reproducible locally. Set `FASTCHECK_SEED` (e.g. in a nightly,
+ * non-merge-gating job) to explore other input families.
+ */
+export const FASTCHECK_SEED = Number(process.env.FASTCHECK_SEED ?? 0x5eed);
```
[the remark this answers is not in the document]
Comment (goanpeca) [reply tests/support/fast-check-seed.ts]: Fixed: resolveFastCheckSeed now falls back to the default on an unset or empty value and throws on any non-integer, so an empty or typo'd FASTCHECK_SEED can never silently pass 0 or NaN to fast-check.
